src/log.js

```diff
@@ -417,6 +418,30 @@ function pushIfNonEmpty(array, val) {
   }
 }
 
+/**
+ * Some exceptions (DOMException, namely) have read-only message.
+ * @param {!Error} error
+ * @return {!Error};
+ */
+export function duplicateErrorIfNecessary(error) {
+  const message = error.message;
+  const test = String(Math.random());
+  error.message = test;
+
+  if (error.message === test) {
+    error.message = message;
+    return error;
+  }
+
+  const e = new Error(error.message);
+  // Copy all the extraneous things we attach.
+  for (const prop in error) {
+    e[prop] = error[prop];
+  }
+  // Ensure these are copied.
+  e.stack = error.stack;
```

Comment:
I'm not 100% sure that `stack` is an enumerable property in all browsers, so it seemed like a good defense.

Or, are you asking about overriding the stack? In that case, yup, it's a thing. It's used a lot to clean stack traces, avoiding "internal" frames.